Sources/SwiftFusion/Inference/ProbablisticTrackingFactor.swift

```diff
@@ -0,0 +1,70 @@
+import PenguinParallel
+import PenguinStructures
+import TensorFlow
+
+extension AppearanceModelEncoder {
+  @differentiable
+  fileprivate func encode<V: FixedSizeVector>(_ image: Tensor<Double>) -> V {
+    V(flatTensor: encode(image.expandingShape(at: 0)).squeezingShape(at: 0))
+  }
+}
+
+/// A factor over a target's pose and appearance in an image.
+public struct ProbablisticTrackingFactor<
+    Encoder: AppearanceModelEncoder,
+    ForegroundModel: GaussianModel, BackgroundModel: GaussianModel
```

Comment:
I think these should not be GaussianModel but "GenerativeDensity"

Reply:
Changed